classes/file_manager.php

```diff
@@ -0,0 +1,186 @@
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Class to manage seb files, used in part to abstract some of the Moodle file manager logic.
+ *
+ * @package    quizaccess_seb
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace quizaccess_seb;
+
+use stored_file;
+
+defined('MOODLE_INTERNAL') || die();
+
+class file_manager {
+
+    /** Component the file will be stored in.  */
+    const COMPONENT = 'mod_quiz';
+
+    /** File area in the module to store files.  */
+    const FILE_AREA = 'quizaccess_seb_quizsettings';
+
+    /**
+     * Get a stored file object representing a draft file saved from a form.
+     *
+     * @param string $itemid
+     * @return stored_file|null
+     * @throws \coding_exception
+     */
+    public function get_form_file_by_itemid(string $itemid) {
+        global $USER; // Uploaded files initially have a user context.
+        $file  = null;
+
+        $fs = get_file_storage();
+        $context = \context_user::instance($USER->id);
+        $areafiles = $fs->get_area_files($context->id, 'user', 'draft', $itemid, 'id DESC', false);
+        if (count($areafiles) === 1) {
+            $file = reset($areafiles);
+        }
+        return $file;
+    }
+
+    /**
+     * Get a saved file from the safe exam browser file area, for a specific quiz.
+     *
+     * @param string $itemid
+     * @param string $cmid
+     * @return stored_file|null
+     * @throws \coding_exception
+     */
+    public function get_module_file_by_itemid(string $itemid, string $cmid) {
+        $file  = null;
+        $fs = get_file_storage();
+
+        if (get_coursemodule_from_id('quiz', $cmid) && !empty($itemid)) { // Validate course module id.
+            $context = \context_module::instance($cmid);
+            $areafiles = $fs->get_area_files($context->id, self::COMPONENT, self::FILE_AREA, $itemid, 'id DESC', false);
+            if (count($areafiles) === 1) {
+                $file = reset($areafiles);
+            }
+        }
+        return $file;
+    }
+
+    /**
+     * Delete a saved file from the safe exam browser file area, for a specific quiz.
+     *
+     * @param string $itemid
+     * @param string $cmid
+     * @return bool
+     * @throws \coding_exception
+     */
+    public function delete_module_file_by_itemid(string $itemid, string $cmid) : bool {
+        $file = $this->get_module_file_by_itemid($itemid, $cmid);
+        if (!empty($file)) {
+            return $file->delete();
```

Comment:
do we want to delete all files if get_module_file_by_itemid potentially return bunch of them?

Reply:
I can't find the docs stating it, but doesn't using the 'itemid' mean that only one file should be returned? Need to confirm this assumption.

`get_module_file_by_itemid` will only return one file, or null.

Reply:
But potentially there could b e more than one file. Do we want to delete them all?